source/common/router/config_impl.h

```diff
@@ -578,6 +578,7 @@ class InternalRedirectPolicyImpl : public InternalRedirectPolicy {
   }
 
   std::vector<InternalRedirectPredicateSharedPtr> predicates() const override;
+  std::vector<Http::LowerCaseString> responseHeadersToCopy() const override;
```

Comment:
That sounded like a great idea (though I doubt it makes much of a difference in practice), but wow, I had to touch a bunch of tests because references don't play as nicely with MOCK_METHOD.

It's done, at least.